json/src/test/kotlin/ktx/json/serializerTest.kt

```diff
@@ -0,0 +1,78 @@
+package ktx.json
+
+import com.badlogic.gdx.utils.Json
+import com.badlogic.gdx.utils.JsonValue
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import io.kotlintest.matchers.shouldEqual
+import io.kotlintest.matchers.shouldThrow
+import org.junit.Test
+
+@Suppress("unused")
+class `should implement ReadOnlySerializer with no 'write' method overriden`<T> : ReadOnlySerializer<T> {
+  // Guarantees that [write] method is optional to implement.
+
+  override fun read(json: Json, jsonData: JsonValue, type: Class<*>?): T {
+    throw NotImplementedError()
+  }
+}
+
+class ReadOnlySerializerTest {
+
+  @Test
+  fun `default implementation for 'write' method should throw UnsupportedOperationException`() {
+    shouldThrow<UnsupportedOperationException> {
+      object : ReadOnlySerializer<Any> {
+        override fun read(json: Json, jsonData: JsonValue, type: Class<*>?): Any = throw NotImplementedError()
+      }.write(Json(), JsonValue(JsonValue.ValueType.`object`), Any::class.java)
+    }
+  }
+}
+
+class ReadOnlySerializerFactoryTest {
+  @Test
+  fun `should return readonly serializer that calls lambda function with 3 parameters`() {
+    val json = Json()
+    val jsonValue = JsonValue(JsonValue.ValueType.`object`)
+    val readFunction: (Json, JsonValue, Class<*>?) -> Data = mock()
+
+    readOnlySerializer(readFunction).read(json, jsonValue, Data::class.java)
+
+    verify(readFunction).invoke(json, jsonValue, Data::class.java)
+  }
+
+  @Test
+  fun `should return readonly serializer that returns result of calling lambda with 3 parameters`() {
+    val data = Data(intField = 10, stringField = "Test", doubleField = 11.254)
+    val json = Json()
+    val jsonData = JsonValue(JsonValue.ValueType.`object`)
+    val readFunction: (Json, JsonValue, Class<*>?) -> Data = { _, _, _ -> data }
+
+    val value = readOnlySerializer(readFunction).read(json, jsonData, Data::class.java)
+
+    value shouldEqual data
+  }
+
+  @Test
+  fun `should return readonly serializer that calls lambda function with 2 parameters`() {
+    val jsonValue = JsonValue(JsonValue.ValueType.`object`)
+    val readFunction: (JsonValue) -> Data = mock()
+
+    readOnlySerializer(readFunction).read(Json(), jsonValue, Data::class.java)
+
+    verify(readFunction).invoke(jsonValue)
+  }
+
+  @Test
+  fun `should return readonly serializer that returns result of calling lambda with 2 parameters`() {
+    val data = Data(intField = 10, stringField = "Test", doubleField = 11.254)
+    val jsonData = JsonValue(JsonValue.ValueType.`object`)
+    val readFunction: (JsonValue) -> Data = { _ -> data }
```

Comment:
In case of a simple parameter lambda, I believe just `{ data }` would work, since the name of the parameter will default to `it`.